### ISSUE TITLE
Mini Cart: Migrate the Mini Cart drawer to use Interactivity API

### ISSUE THREAD

**Issue by samueljseay** [woothemes/woocommerce#51080]
This is a follow up task to #51078 

Once that task is done, we can start to replace the functionality of the opening/closing drawer that contains items in cart to use the Interactivity API.

Note, out of scope: The inner blocks of the drawer would at least for now still be rendered with React in this stage, but in a later stage we would use Interactivity API directives to render those components with Preact instead.
